src/utils/validation.ts

```
@@ -0,0 +1,129 @@
+  if (!(map instanceof Map)) {
+    return { valid: false, errors: [{ field: 'morphologyMap', message: 'Must be a Map' }] };
+  }
+
+  const errors: ValidationError[] = [];
+
+  map.forEach((entry, key) => {
+    if (typeof key !== 'number') {
+      errors.push({ field: 'morphologyMap', message: `Key must be a number, got: ${typeof key}` });
+    }
+    if (!Array.isArray(entry?.lemmas)) {
+      errors.push({ field: 'morphologyMap.lemmas', message: `Entry ${key}: lemmas must be an array` });
```

ESLint

[error] Replace `·field:·'morphologyMap.lemmas',·message:·`Entry·${key}:·lemmas·must·be·an·array`` with `⏎········field:·'morphologyMap.lemmas',⏎········message:·`Entry·${key}:·lemmas·must·be·an·array`,⏎·····`

(prettier/prettier)